src/lib/mongoBackend/mongoListSubscriptions.cpp

```diff
@@ -23,42 +23,171 @@
 * Author: Orion dev team
 */
 
+#include "common/sem.h"
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
 #include "mongoBackend/mongoListSubscriptions.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "mongoBackend/collectionOperations.h"
+
+#include "mongo/client/dbclient.h"
+
+using namespace ngsiv2;
+
+/* ****************************************************************************
+*
+* setSubscriptionId -
+*/
+extern void setSubscriptionId(Subscription* s, const BSONObj& r)
+{
+  s->id = r.getField("_id").OID().toString();
```

Comment:
`_id` is a mandatory field in every document in every collection in MongoDB. Only if the driver fails or some other very critical and werid situation occur we will get here a SGIEGV.

Somehow related and more dangerous (as MongoDB doesn't have any "protection" in this case) is not finding a field in the returned BSON or having the field but with a different type (e.g. it is a string but mongoBackend code expect and integer). The fix for this is included in https://github.com/telefonicaid/fiware-orion/issues/136